dragon.c

```diff
@@ -137,8 +141,51 @@ void drag_end(GtkWidget *widget, GdkDragContext *context, gpointer user_data) {
         if (action_str[0] == 'i')
             free(action_str);
     }
-    if (and_exit)
+
+    if (quit_mode == QUIT_ALL){
         gtk_main_quit();
+    }
+
+    if (quit_mode == QUIT_ITEM) {
```

Comment:
Space after `)`. Also since they are mutually exclusive, `else if` makes more sense IMO.

```suggestion
    if (quit_mode == QUIT_ALL) {
        gtk_main_quit();
    } else if (quit_mode == QUIT_ITEM) {
```